backend/src/main/java/ca/bc/gov/hlth/hnweb/service/AuditService.java

```diff
@@ -126,7 +126,7 @@ public Transaction createTransaction(String sourceIP, TransactionType type) {
 		transaction.setServer(getServer());
 		transaction.setSessionId(userInfo != null ? userInfo.getSessionState() : null);
 		transaction.setSourceIp(sourceIP);
-		transaction.setSpgRole(SecurityUtil.loadSPGBasedOnTransactionType(userInfo, type));
+		transaction.setSpgRole(userInfo != null ? SecurityUtil.loadSPGBasedOnTransactionType(userInfo, type) : null);
```

Comment:
Are there changes that can be made to SecurityUtil.loadUserInfo() so that the UserInfo can actually be created in this scenario? I don't that code should fail just because there are no roles.

Reply:
Changes were made.